BackendTaskPilot/Program.cs

```
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
+        var response = JsonSerializer.Deserialize<CnbRatesResponse>(data);
+
+        return response.Rates.Select(rate => rate).Where(rate => currencies.Contains(rate.CurrencyCode));
```

No need for `Select(rate => rate)`, simplify to `return response.Rates.Where(rate => currencies.Contains(rate.CurrencyCode));`

---

This now returns any exchange rates found that are also within the desired currency list. Consider the case of the request of currencies being the full USD, EUR, JPY, SAR, KES, but CNB only returns USD, EUR, JPY exchange rate mappings. What is the desired outcome then? 
If we desire to only return what was found, I think we should still add some logging to express a warning that the request was not able to be fulfilled in whole. It should also be documented for the endpoint that it is a best-effort. If we instead choose to handle it in full, we should return a failure to the client with a message stating which currencies could not be given rates for and thus caused the failure.

---

We should also consider the case of the empty target currencies list. What are the scenarios this endpoint will be used for. 
Would it make sense to allow an empty target list to be interpreted as no filtering of the result, i.e return all exchange rates retrievable of the 180 possible? 
Or should we keep it as is now, no target currency -> no result exchange rates? If so we can validate list size earlier in this method and skip the call to CNB.